addon/-private/system/record-arrays/record-array.js

```diff
@@ -88,11 +79,23 @@ export default Ember.ArrayProxy.extend(Ember.Evented, {
   },
 
   replace() {
-    let type = get(this, 'type').toString();
-    throw new Error(`The result of a server query (for all ${type} types) is immutable. To modify contents, use toArray()`);
+    throw new Error(`The result of a server query (for all ${this.modelName} types) is immutable. To modify contents, use toArray()`);
   },
 
   /**
+   The modelClass represented by this record array.
+
+   @property type
+   @type DS.Model
+   */
+  type: computed('modelName', function() {
+    if (!this.modelName) {
+      return null;
+    }
+    return this.store.modelFor(this.modelName);
+  }),
```

Comment:
This doesn't do what you think it is. (the last arg is dropped, but i suspect we want it to actually run)

I suspect you want:

```js
type: computed('modelName', function() {
  // ...
}).readOnly()
```